packages/neuron-wallet/src/services/transaction-sender.ts

```diff
@@ -54,20 +54,30 @@ export default class TransactionSender {
 
     await TransactionPersistor.saveSentTx(tx, txHash)
 
-    await WalletService.checkAndGenerateAddresses(walletID)
+    const wallet = WalletService.getInstance().get(walletID)
+    await wallet.checkAndGenerateAddresses()
     return txHash
   }
 
   private async sign(walletID: string = '', transaction: Transaction, password: string = '', skipLastInputs: number = 0) {
     const wallet = this.walletService.get(walletID)
-
-    if (password === '') {
-      throw new IsRequired('Password')
-    }
-
     const tx = Transaction.fromObject(transaction)
     const { ckb } = NodeService.getInstance()
     const txHash: string = tx.computeHash()
+    if (wallet.isHardware()) {
```

Comment:
Do we have tests covering this part?

Reply:
No, I plan to implement the HD wallet feature and then add test cases.

Reply:
The HD wallet will be implemented in the next iteration. So I would suggest adding the necessary tests for the changes that have been done.